dandi/dandiarchive.py

```diff
@@ -526,8 +526,8 @@ def follow_redirect(url):
         i = 0
         while True:
             r = requests.head(url, allow_redirects=True)
-            if r.status_code == 404:
-                if i < 3:
+            if r.status_code in (404, 502, 503, 504):
```

Comment:
```suggestion
            if r.status_code in (400, 502, 503, 504):
```
per brief investigation in https://github.com/dandi/dandi-cli/issues/261#issuecomment-909631735